### DISCUSSION TITLE
Image upload as datauri - base64

### DISCUSSION THREAD

**Discussion by jimdaro** [Budibase/budibase#5766] [Help]
Hi Folks!

Any idea how to make it in Budibase? I use an api which accepts image data as data uri but I'm can't get BB to generate it with custom javascript codes.

Thanks!

JD

**Comment by melohagan** [comment]
Hey @jimdaro 

Are you trying to grab the image data from a file attachment? 
The images that you upload as file attachments, or through the image component use URLs. So there is no binding to the underlying image data. 

What's your use case?

**Comment by jimdaro** [comment]
Hi @melohagan!

Thanks for replying. I'm trying to upload an image to a third party API but it's only accepting datauri - base64 coded. I try to make a custom javascript code to fetch the image from the url and transform it, but I can't get BB to do it, I always recevie errors. :(

I mean something like this.

`
function toDataURL(src, callback) {
var img = new Image();
img.crossOrigin = 'Anonymous';
img.onload = function() {
    var canvas = document.createElement('CANVAS');
    var ctx = canvas.getContext('2d');
    var dataURL;
    canvas.height = this.naturalHeight;
    canvas.width = this.naturalWidth;
    ctx.drawImage(this, 0, 0);
    dataURL = canvas.toDataURL();
    callback(dataURL)
  };
  img.src = src;
  if (img.complete || img.complete === undefined) {
    img.src = "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///ywAAAAAAQABAAACAUwAOw==";
    img.src = src;
  ;
  }
}
toDataURL('https://unsplash.it/960/540?random', function(dataUrl) { console.log(dataUrl); return dataUrl;})
`

**Reply by melohagan** [reply]
You could maybe try using an Embed to get the JavaScript to work, for example: https://github.com/Budibase/budibase/discussions/4433#discussioncomment-2701647

Otherwise you could access the Minio bucket (https://github.com/Budibase/budibase/discussions/5084) to get the files and use a third party site (https://base64.guru/converter/encode/image/gif) and perhaps a bash script to encode them.
There should be an API for Minio as well.  I don't have a handy solution for this unfortunately.
